src/managers/config.py

```diff
@@ -186,6 +188,32 @@ def heap_opts(self) -> str:
 
         return f"KAFKA_HEAP_OPTS='{' '.join(opts)}'"
 
+    @property
+    def security_protocol(self) -> AuthMechanism:
+        """Infers current charm security.protocol based on current relations."""
+        # FIXME: When we have multiple auth_mechanims/listeners, remove this method
+        return (
+            "SASL_SSL"
+            if (self.state.cluster.tls_enabled and self.state.unit_broker.certificate)
+            else "SASL_PLAINTEXT"
+        )
```

Comment:
This should be moved to `state`, and just infer the ports from the security.protocol state here.